### ISSUE TITLE
Atlas 2.6 - timeouts retrieving JavaScript dependencies

### ISSUE THREAD

**Issue by anthonysena** [OHDSI/Atlas#1312]
### Known bug with ATLAS 2.6

We've experienced issues when ATLAS 2.6 loads that not all JavaScript modules are properly downloaded from content delivery networks (CDNs). In particular, this problem was noted for resources loaded off of the UNPKG CDN (https://unpkg.com/). This issue will describe a work-around that you can use to address the problem since we've also discovered that the build mechanism that was introduced in ATLAS 2.5 to download and bundle all JavaScript dependencies is not properly functioning in ATLAS 2.6 but has been fixed for the planned ATLAS 2.7 release.

### Work-around steps

Clone the ATLAS repository from the current master branch (**_do not do this in a location where you are currently hosting ATLAS on a web server_**). Once downloaded, open an NPM command line in the ATLAS root directory and run:

`npm run build`

This will retrieve all of the JavaScript dependencies required for 2.6 and put them into a folder called `node_modules`. Copy these to the root of your ATLAS 2.6 folder where you have ATLAS on a web server.

Next, modify the `paths` section of `js/settings.js` file to replace the CDN references to the `node_modules` folder. Here is an example:

#### Before

`"ohdsi-api": "https://unpkg.com/@ohdsi/ui-toolbox@1.0.2/lib/umd/api/index"`

#### After

`"ohdsi-api": "../node_modules/@ohdsi/ui-toolbox/lib/umd/api/index"`

Refresh your browser and confirm that the JavaScript modules now load properly and double-check your paths to ensure you are pointed to the proper location in the `node_modules` folder.

Please leave any questions/clarifications/issues on this thread. Thanks!

**Comment by pavgra** [comment]
@anthonysena , what's wrong with the bundling script built into 2.6.0? why would you need workarounds?

**Comment by t-abdul-basser** [comment]
Thanks for noting this @anthonysena. I was just debugging this. This is an issue that has been causing us significant grief on several deployments here for a few days now :)

**Comment by anthonysena** [comment]
@pavgra - When attempting this yesterday off of the 'released' branch, I was getting an error:

`Error: ENOENT: no such file or directory, open 'Atlas/js/assets/bundle/cdn-libs/atlascharts.js'`

Tracing through it, I found that this bit of the code was having an issue for those packages stored up on UNPKG.org: https://github.com/OHDSI/Atlas/blob/released/build/index.js#L31. Only the libraries on the CDN failed to download and thus the bundling failed. Since I know this is fixed on master, I used this work-around for now. If you have ideas on how to fix it, I'm open to it - just didn't want to spend a lot of time debugging this.
